packages/hardhat-etherscan/src/index.ts

```diff
@@ -311,3 +406,46 @@ task("verify", "Verifies contract on etherscan")
     []
   )
   .setAction(verify);
+
+const getMinimumBuild: ActionType<MinimumBuildArgs> = async function (
+  { sourceName },
+  { config, run }
+) {
+  const dependencyGraph: DependencyGraph = await run(
+    TASK_COMPILE_SOLIDITY_GET_DEPENDENCY_GRAPH,
+    { sourceNames: [sourceName] }
+  );
+
+  const resolvedFiles = dependencyGraph
+    .getResolvedFiles()
+    .filter(({ sourceName: thisSourceName }) => {
+      return thisSourceName === sourceName;
+    });
+
```

Comment:
Maybe it's worth it checking here that `resolvedFiles.length` is 1. You can use `assertHardhatInvariant` for this.

Reply:
I ended up implementing a sort of equivalent `assertHardhatPluginInvariant`.